caravel/assets/javascripts/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -157,14 +181,15 @@ class SqlEditorLeftBar extends React.PureComponent {
           />
         </div>
         <div className="m-t-5">
-          <Select
+          <Select.Async
```

Comment:
Yep, it does roundtrip once per key stroke, caching handles it well, didn't see any lags, happy to show you on my laptop. 
There is option to pass `complete: true`in async options, there were few bugs around that.